core/platform/src/error_code.rs

```diff
@@ -1,25 +1,36 @@
-/// An error code returned by the kernel. Tock's system calls return errors as a
-/// negative `isize`. This wraps the isize, and is useful for adding type safety
-/// to APIs.
-
+/// A system call error code. This can either be an error code returned by the
+/// kernel or BADRVAL, which indicates the kernel returned the wrong type of
+/// response to a system call.
+// ErrorCode is not an enum so that conversion from the kernel's return value (a
+// `usize` in a register) is free.
 #[derive(Clone, Copy, PartialEq, Eq)]
 pub struct ErrorCode {
```

Comment:
I'm curious why you chose to make it vanilla struct instead of a [tuple struct](https://doc.rust-lang.org/1.9.0/book/structs.html#tuple-structs), as I'm more used to seeing wrappers this way. This, of course, is entirely up to you.

Reply:
I don't really like tuple structs because I don't think the `.0`/`.1`/etc naming scheme is a good idea, so I don't really think of them.

However, tuple structs are the idiomatic Rust way to implement wrapper types. I'll go ahead and make that change, as I think being idiomatic is more important than my personal preference here.